belgif-rest-problem-spring/src/main/java/io/github/belgif/rest/problem/spring/OpenApiRequestViolationProblemAdvice.java

```diff
@@ -0,0 +1,198 @@
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.core.annotation.Order;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+import com.atlassian.oai.validator.report.ValidationReport;
+import com.atlassian.oai.validator.springmvc.InvalidRequestException;
+import com.fasterxml.jackson.core.JsonPointer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.github.belgif.rest.problem.BadRequestProblem;
+import io.github.belgif.rest.problem.ResourceNotFoundProblem;
+import io.github.belgif.rest.problem.api.InEnum;
+import io.github.belgif.rest.problem.api.InputValidationIssue;
+import io.github.belgif.rest.problem.api.Problem;
+import io.swagger.v3.oas.models.parameters.Parameter;
+
+@RestControllerAdvice
+@ConditionalOnWebApplication
+@Order(1)
+@ConditionalOnClass(InvalidRequestException.class)
+public class OpenApiRequestViolationProblemAdvice {
+
+    private static final URI SCHEMA_VIOLATION_URI =
+            URI.create("urn:problem-type:belgif:input-validation:schemaViolation");
+    private static final String ISSUE_TITLE = "Input value is invalid with respect to the schema.";
+    private static final Logger log = LoggerFactory.getLogger(OpenApiRequestViolationProblemAdvice.class);
+
+    private final ObjectMapper mapper;
+
+    public OpenApiRequestViolationProblemAdvice(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    @ExceptionHandler(value = InvalidRequestException.class)
+    public ResponseEntity<Problem> handleInvalidRequestException(final InvalidRequestException ex,
+            HttpServletRequest request) {
+        if (isNonExistingPath(ex)) {
+            return ProblemMediaType.INSTANCE.toResponse(new ResourceNotFoundProblem());
+        }
+        List<InputValidationIssue> issues = constructIssues(ex, request);
+        return ProblemMediaType.INSTANCE.toResponse(new BadRequestProblem(issues));
+    }
+
+    private List<InputValidationIssue> constructIssues(InvalidRequestException ex, HttpServletRequest request) {
+        Set<InputValidationIssue> issues = new HashSet<>();
+        for (ValidationReport.Message message : ex.getValidationReport().getMessages()) {
+            InputValidationIssue issue = handleSpecialCases(message);
+            if (issue != null) {
+                issues.add(issue);
+                continue;
+            }
+            issue = getBasicIssue(message);
```

Comment:
Can you use the existing factory method for constructing the InputValidationIssue?

https://github.com/belgif/rest-problem-java/blob/9cfda64728f64775b2765ddd87dbaeba1a36fe6c/belgif-rest-problem/src/main/java/io/github/belgif/rest/problem/api/InputValidationIssues.java#L61-L66